datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -497,6 +503,37 @@ def load_configuration_models(self, package_path=None):
             if instance_config is not None:
                 self._config_model_instance = instance_config
 
+    def load_memray_context_manager(self):
+        enable_memray = is_affirmative(self.instance.get('enable_memray', self.init_config.get('enable_memray', False)))
+
+        if enable_memray:
+            if PY2:
+                raise ConfigurationError("`enable_memray` option is not supported for py2 environments.")
+
+            if sys.platform not in ("linux", "darwin"):
+                raise ConfigurationError("`enable_memray` option is only supported on Linux and macOS.")
+
+            output_file = self.instance.get('memray_file', self.init_config.get('memray_file'))
+
+            # Append the check_id if the name if globally specified
+            if self.init_config.get('memray_file') and not self.instance.get('memray_file'):
+                output_file += ".{}".format(self.check_id)
+
+            native_mode = self.instance.get('memray_native_mode', self.init_config.get('memray_native_mode', False))
+
+            if output_file is None:
+                raise ConfigurationError('Setting `memray_file` must be provided to use memray.')
+
+            import memray
+
+            # Default to 1h if min_collection_interval is set to 15s (the default value)
+            self._memray_remaining_iterations = self.instance.get(
+                'memray_iteration_count', self.init_config.get('memray_iteration_count', 240)
+            )
```

Comment:
Would we really need to explicitly stop memray? It would look harmless to me to skip the exit, judging by https://github.com/bloomberg/memray/blob/f9fbe27c3750e29ac40f6c10498aa6d661661f08/src/memray/_memray.pyx#L420-L423. It would simplify the whole approach.